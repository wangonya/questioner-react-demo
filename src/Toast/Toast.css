.err {
  visibility: hidden;
  min-width: 250px;
  background-color: #f0776c;
  color: white;
  text-align: center;
  border-radius: 5px;
  padding: 15px;
  position: fixed;
  z-index: 1;
  left: 38%;
  top: 30px;
  visibility: visible;
  -webkit-animation: fadein 0.5s, fadeout 0.5s 3.5s;
  animation: fadein 0.5s, fadeout 0.5s 3.5s;
}

.ok {
  min-width: 250px;
  background-color: #53e3a6;
  color: white;
  text-align: center;
  border-radius: 5px;
  padding: 15px;
  position: fixed;
  z-index: 1;
  left: 38%;
  top: 30px;
  visibility: visible;
  -webkit-animation: fadein 0.5s, fadeout 0.5s 3.5s;
  animation: fadein 0.5s, fadeout 0.5s 3.5s;
}

@-webkit-keyframes fadein {
  from {
    top: 0;
    opacity: 0;
  }
  to {
    top: 30px;
    opacity: 1;
  }
}

@keyframes fadein {
  from {
    top: 0;
    opacity: 0;
  }
  to {
    top: 30px;
    opacity: 1;
  }
}

@-webkit-keyframes fadeout {
  from {
    top: 30px;
    opacity: 1;
  }
  to {
    top: 0;
    opacity: 0;
  }
}

@keyframes fadeout {
  from {
    top: 30px;
    opacity: 1;
  }
  to {
    top: 0;
    opacity: 0;
  }
}
