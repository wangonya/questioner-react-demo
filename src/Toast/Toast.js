import React from "react";

import "./Toast.css";

const Toast = props => {
  return <div className={props.type}>{props.message}</div>;
};

export default Toast;
