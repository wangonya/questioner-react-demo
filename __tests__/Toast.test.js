import React from "react";
import { shallow, render } from "enzyme";

import Toast from ".././src/Toast/Toast";

describe("<Toast /> component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Toast />);
  });

  test("renders without crashing -- no props", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div.ok")).toHaveLength(0);
  });

  test("renders without crashing -- with props", () => {
    wrapper.setProps({ type: "ok", message: "testing" });
    expect(wrapper.find("div.ok")).toHaveLength(1);
    expect(wrapper.text()).toBe("testing");
  });
});
