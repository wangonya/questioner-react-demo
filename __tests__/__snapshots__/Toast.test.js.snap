// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> component renders without crashing -- no props 1`] = `ShallowWrapper {}`;
